src/MainWindow/main_window.cpp

```diff
@@ -345,6 +345,19 @@ void MainWindow::addPinPlannerRefreshButton(QDockWidget* dock) {
   QWidget* w = new QWidget;
   auto layout = new QHBoxLayout;
   layout->addWidget(new QLabel{dock->windowTitle()});
+
+  auto saveButton = new QPushButton{dock};
+  saveButton->setObjectName("saveButton");
+  connect(saveButton, &QPushButton::clicked, this,
+          &MainWindow::saveActionTriggered);
+  saveButton->setSizePolicy(
+      QSizePolicy{QSizePolicy::Maximum, QSizePolicy::Maximum});
+  saveButton->setIcon(QIcon{":/images/save-action.png"});
```

Comment:
Can't we use the same save icon? Also, where did we get this from?

Reply:
I've got it form https://iconoir.com/. I don't know where we got the previous. I think it make sense to replace it.

Reply:
We will have to bring in it's github codebase as a third-party module, let's talk offline.

Reply:
Do you want me to use same icon?

Reply:
No

Reply:
A reference to the source is also acceptable.

Reply:
@KochynVolodymyr lets add a link here for reference as Alain proposed and will merge in later tonight. I will work offline to get other licenses pulled in.

Reply:
Link: data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz48c3ZnIHdpZHRoPSIyNHB4IiBoZWlnaHQ9IjI0cHgiIHN0cm9rZS13aWR0aD0iMS41IiB2aWV3Qm94PSIwIDAgMjQgMjQiIGZpbGw9Im5vbmUiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgY29sb3I9IiMyMDRhODciPjxwYXRoIGQ9Ik0zIDcuNVY1YTIgMiAwIDAxMi0yaDExLjE3MmEyIDIgMCAwMTEuNDE0LjU4NmwyLjgyOCAyLjgyOEEyIDIgMCAwMTIxIDcuODI4VjE5YTIgMiAwIDAxLTIgMkg1YTIgMiAwIDAxLTItMnYtMi41TTYgMjF2LTQiIHN0cm9rZT0iIzIwNGE4NyIgc3Ryb2tlLXdpZHRoPSIxLjUiIHN0cm9rZS1saW5lY2FwPSJyb3VuZCIgc3Ryb2tlLWxpbmVqb2luPSJyb3VuZCI+PC9wYXRoPjxwYXRoIGQ9Ik0xOCAyMXYtNy40YS42LjYgMCAwMC0uNi0uNkgxNU0xNiAzdjUuNGEuNi42IDAgMDEtLjYuNmgtMS45TTggM3YzTTEgMTJoMTFtMCAwTDkgOW0zIDNsLTMgMyIgc3Ryb2tlPSIjMjA0YTg3IiBzdHJva2Utd2lkdGg9IjEuNSIgc3Ryb2tlLWxpbmVjYXA9InJvdW5kIiBzdHJva2UtbGluZWpvaW49InJvdW5kIj48L3BhdGg+PC9zdmc+

Reply:
Here is link for download:
data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz48c3ZnIHdpZHRoPSIyNHB4IiBoZWlnaHQ9IjI0cHgiIHN0cm9rZS13aWR0aD0iMS41IiB2aWV3Qm94PSIwIDAgMjQgMjQiIGZpbGw9Im5vbmUiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgY29sb3I9IiMyMDRhODciPjxwYXRoIGQ9Ik0zIDE5VjVhMiAyIDAgMDEyLTJoMTEuMTcyYTIgMiAwIDAxMS40MTQuNTg2bDIuODI4IDIuODI4QTIgMiAwIDAxMjEgNy44MjhWMTlhMiAyIDAgMDEtMiAySDVhMiAyIDAgMDEtMi0yeiIgc3Ryb2tlPSIjMjA0YTg3IiBzdHJva2Utd2lkdGg9IjEuNSI+PC9wYXRoPjxwYXRoIGQ9Ik04LjYgOWg2LjhhLjYuNiAwIDAwLjYtLjZWMy42YS42LjYgMCAwMC0uNi0uNkg4LjZhLjYuNiAwIDAwLS42LjZ2NC44YS42LjYgMCAwMC42LjZ6TTYgMTMuNlYyMWgxMnYtNy40YS42LjYgMCAwMC0uNi0uNkg2LjZhLjYuNiAwIDAwLS42LjZ6IiBzdHJva2U9IiMyMDRhODciIHN0cm9rZS13aWR0aD0iMS41Ij48L3BhdGg+PC9zdmc+

Reply:
Thanks. I will work on integrating that github codebase separately, will merge this in.